core/src/wsv.rs

```diff
@@ -487,6 +489,12 @@ impl WorldStateView {
         &self.world.roles
     }
 
+    /// Get all permission token ids
+    #[inline]
+    pub fn permission_token_ids(&self) -> &crate::PermissionTokenIds {
+        &self.world.permission_token_ids
+    }
+
```

Comment:
I understand, that this looks reasonable, but I just don't like that anyone can modify `PermissionTokenIds` by non-mutable reference (because it's the `DashSet`).

@appetrosyan , @mversic ,  what do you think? Is this okay? Or maybe someday we can refactor `WorldStateView` to remove things like that using iterators or *inspect*-like approach?

Reply:
Unfortunately any concurrent (high-performance) `Set` implementation will have to pass a non-`mut` reference, and so anyone holding it can modify. The only way of preventing others from modifying it, is to expose a subset of its functionality either via a wrapper `struct`, or by completely hiding the `DashSet` and instead using `inspect`. 

I don't think that this is such a huge problem, though. In e.g. Java the point is that you want to be Open-Closed. In rust the same approach doesn't work, because you have to statically link against the other libraries' source code, and we're writing open source software. 

Thus, we shouldn't try to defend against malicious modification (we have the Sumeragi algorithm for that), but instead focus on preventing user error. In this case, modifying the `permission_token_ids` from outside the blockchain is a mistake, which can be fixed by returning a `PermissionTokenIds` wrapper, which can be read as a `DashMap` but not modified. Ideally, for FFI, it might even be good to write it as a `Vec` of tuples.

Reply:
@appetrosyan , I understand, but I don't mean external users. I mean developers and accidental mistakes. I just want to improve `WSV` design so that modification won't be possible if the method is not designed for it